adapters/andalucia.js

```diff
@@ -0,0 +1,925 @@
+      }
+      res = res[0].concat(res[1]);
+      done(null, res);
+    });
+  };
+};
+
+const formatData = ($, stations) => {
+  const tables = [];
+  // the tables we want don't have cellpadding 5.
+  $('table').each((i, el) => {
+    const table = $(el)['0'];
+    if (table.attribs.cellpadding !== '5') {
+      tables.push($(el));
+    }
+  });
+  // data is held two consecutive tables
+  // (one with station name, other actual data)
+  let stationData = _.chunk(tables, 2);
+  // return a list of measurement objects for each table.
+  return stationData.map((table) => {
+    const station = $(table[0]['0'].children[2]).text().split('Estacion')[1].trim();
+    const stationLoc = mapStationCoords(station);
+    table = makeTable(table[1]['0'], $);
+    return makeMeasurements(table, stationLoc);
+  });
+};
+
+const mapStationCoords = (station) => {
+  return stations.filter((stationObj) => {
+    return stationObj['name'] === station;
+  }).map((stationObj) => {
+    return {
+      longitude: stationObj['coordinates'][0],
+      latitude: stationObj['coordinates'][1]
+    };
+  });
+};
+
+const makeTable = (cheerioTable, $) => {
+  return cheerioTable.children.filter((child) => {
+    return child.name === 'tr';
+  }).map((row) => {
+    const tds = [];
+    $(row).children().each((i, el) => {
+      tds.push($(el).text());
+    });
+    return tds;
+  });
+};
+
+const makeMeasurements = (stationData, stationLoc) => {
+  // get index of data sources we can record
+  const pollutantIndexes = stationData[0].filter((cell) => {
+    return _.includes(['SO2', 'NO2'], cell);
+  }).map((validCell) => {
+    return stationData[0].indexOf(validCell);
+  });
+  // include index for 1st column, which has the date
+  pollutantIndexes.push(0);
+  // filter rows for only the date,SO2, and NO2 rows
+  stationData = stationData.map((row) => {
+    return row.filter((cell, index) => {
+      if (_.includes(pollutantIndexes, index)) {
+        return cell;
+      }
+    });
+  });
+  // make each row into a list of measurements
+  stationData = stationData.slice(1, -1).map((row, index) => {
+    return row.map((cell, index) => {
+      if (index !== 0) {
+        return {
+          parameter: stationData[0][index],
+          date: makeDate(row[0]),
+          coordinates: stationLoc,
+          value: cell,
+          unit: 'µg/m³',
+          attribution: [{
+            name: 'Ministry of Environment and Spatial Planning',
+            url: 'http://www.juntadeandalucia.es/medioambiente/site/portalweb'
+          }],
+          averagingPeriod: {
+            unit: 'hours',
+            // averaging period gathered from source.url
+            value: stationData[0][index] === 'SO2' ? 24 : 1
+          }
+        };
+      }
+    }).filter((cell) => {
+      if (cell !== undefined) {
+        return cell;
+      }
+    });
+  });
+  // merge all measurement lists into one
+  return [].concat.apply([], stationData);
+};
+
+const makeDate = (date) => {
```

Comment:
If I'm understanding this correctly, all of this is to get it in a format `moment` can handle? `moment` can be given any arbitrary string pattern, you just need to tell it what it is. So I think this could be greatly simplified.